daemon/cmd/devices.go

```diff
@@ -283,37 +287,59 @@ func (dm *DeviceManager) updateDevicesFromRoutes(l3DevOK bool, routes []netlink.
 // expandDevices expands all wildcard device names to concrete devices.
 // e.g. device "eth+" expands to "eth0,eth1" etc. Non-matching wildcards are ignored.
 func expandDevices() error {
+	expandedDevices, err := expandDeviceWildcards(option.Config.Devices, option.Devices)
+	if err != nil {
+		return err
+	}
+	option.Config.Devices = expandedDevices
+	return nil
+}
+
+// expandDirectRoutingDevice expands all wildcard device names to concrete devices and picks a first one.
+func expandDirectRoutingDevice() error {
+	if option.Config.DirectRoutingDevice == "" {
+		return nil
+	}
+	expandedDevices, err := expandDeviceWildcards([]string{option.Config.DirectRoutingDevice}, option.DirectRoutingDevice)
+	if err != nil {
+		return err
+	}
+	option.Config.DirectRoutingDevice = expandedDevices[0]
```

Comment:
We should panic if there is more than one device which matches the wildcard direct routing device match. Otherwise, it might lead to situations in which other than expected device has been selected.

Reply:
Thank you for your comment!

It doesn't work at least in our use case if cilium-agent panic if there is more than one device that matches the wildcard direct routing device match.

We have two types of nodes and one has devices eno1 and eno2, another has devices eno8303 and eno8403. In this case, we specify directRoutingDevice=eno+ and expect cilium-agent to pick eno1 or eno8303 accordingly.

I understand your concern. How about printing a warning log and showing users which device has been selected as the direct routing device if there is more than one device that matches the wildcard.

Reply:
@brb What do you think about this?

Reply:
We are currently busy with the final bits of the upcoming v1.11 release. Let us back to you next week.